src/l0_sampling/sketch.cpp

```diff
@@ -55,13 +55,13 @@ Sketch::Sketch(const Sketch& s) : seed(s.seed) {
 }
 
 void Sketch::update(const vec_t& update_idx) {
-  XXH64_hash_t update_hash = Bucket_Boruvka::index_hash(update_idx, seed);
+  vec_hash_t update_hash = Bucket_Boruvka::index_hash(update_idx, seed);
   Bucket_Boruvka::update(bucket_a[num_elems - 1], bucket_c[num_elems - 1], update_idx, update_hash);
   for (unsigned i = 0; i < num_buckets; ++i) {
     col_hash_t col_index_hash = Bucket_Boruvka::col_index_hash(update_idx, seed + i);
     for (unsigned j = 0; j < num_guesses; ++j) {
       unsigned bucket_id = i * num_guesses + j;
-      if (Bucket_Boruvka::contains(col_index_hash, 2 << j)){
+      if (Bucket_Boruvka::contains(col_index_hash, ((col_hash_t)1) << j)){
```

Comment:
why do we change from 2^{j-1} to 2^j?

Reply:
Because the contains query changed from `hash % 2^x == 0` to `hash & x == 0`. In the first version we need j to be one larger than the index we're querying while in the second we don't